### ISSUE TITLE
dag/walk Hangs On Refresh

### ISSUE THREAD

**Issue by WhiskyTest** [opsgenie/terraform-provider-opsgenie#112]
### Terraform Version
Terraform v0.12.26
+ provider.opsgenie v0.3.4

### Affected Resource(s)
- opsgenie_user

### Terraform Configuration Files
```hcl
main.tf
# Configure the Opsgenie Provider
provider "opsgenie" {
  api_key = "xxxxxxx"
  api_url = "api.opsgenie.com"
}

users.tf

resource "opsgenie_user" "user_name" {
  full_name = "User Name"
  locale    = "en_GB"
  role      = "Owner"
  timezone  = "Europe/London"
  username  = "username@subdomain.domain"
}
... 88x users like this
```

### Debug Output
https://gist.github.com/WhiskyTest/521b9075672706119b4650f3e75838f3

### Panic Output
na

### Expected Behavior
Plan should finish running

### Actual Behavior
Plan hangs indefinitely

### Steps to Reproduce
1. Import over 29 existing users (88 in this case) into terraform state: `terraform import opsgenie_user.user_name <id>`
1. `terraform plan`

### Important Factoids

- I've imported 88 users from an existing Opsgenie account into terraform state in order to manage them with terraform.

- Plan will work correctly if I import up to 29 users. But it seems to hang once I go over that number.
- It hangs on different users each time.
- Line 8958 is where I 'end process' on terraform to stop it running, otherwise it will be stuck indefinitely.
### References
na

**Comment by oskarm93** [comment]
I was helping OP to diagnose this issue. Some extra hanging behaviours I saw:
- Running `terraform plan` with a mix of teams and schedules would hang. It would also show `waiting on X` as the last messages in the log. This did not happen with only teams or only schedules defined in the tfstate.
- Running with `-parallelism=1` or not made no difference to hanging / getting stuck.

Steps to reproduce:
- Have about 15 schedules defined in config and tfstate with hard-coded inputs.
- Have about 15 teams defined in config and tfstate with hard-coded inputs.
- Run terraform plan with parallelism=1.
- All schedules will be traversed without a problem and state will be refreshed.
- First team resource will refresh state without an issue.
- Second team resource will hang refreshing state.

I managed to reproduce the issue on Windows by cloning latest master of this repository and building my own executable with `go install` much like in the Makefile. Refresh would always hang in this file:
`https://github.com/terraform-providers/terraform-provider-opsgenie/blob/master/opsgenie/resource_opsgenie_team.go` line 99.

Since this is a dependency call on Opsgenie Go SDK, I cloned the specific version used by the provider and configured build process to use my local copy (in `go.mod`) as a dependency with extra print statements.
```
replace github.com/opsgenie/opsgenie-go-sdk-v2 => <my path>/go/src/github.com/opsgenie/opsgenie-go-sdk-v2@v1.0.7-0.20200406122459-ce3f70f22ab7
```

From there I noticed that this call would actually hang on this file:
`https://github.com/opsgenie/opsgenie-go-sdk-v2/blob/master/client/client.go` line 250.

Terraform provider does not include a Logger by default in its config passed to Opsgenie Client, so this file makes one on the first call, sets log levels and tries to print a statement above multiple times. It will then deadlock or hang printing that statement. As a workaround and confirmation to my hypothesis, I commented out that line in `client.go` and I was able to execute the terraform plan without fail 15/15 times. Putting the line back in, the first plan hangs.

For our case to start using the provider, I made the following workaround in the file below (set log level to Warning so the hanging print statement doesn't execute):
`https://github.com/terraform-providers/terraform-provider-opsgenie/blob/master/opsgenie/config.go`

```
func (c *Config) Client() (*OpsgenieClient, error) {
	myLogger := logrus.New()
	myLogger.SetLevel(logrus.WarnLevel)
	myLogger.SetFormatter(&logrus.JSONFormatter{
			TimestampFormat:  time.RFC3339Nano,
			PrettyPrint:      true,
	},)

	config := &client.Config{
		ApiKey:         c.ApiKey,
		RetryCount:     10,
		OpsGenieAPIURL: client.ApiUrl(c.ApiUrl),
		Backoff:        retryablehttp.DefaultBackoff,
		Logger: myLogger,
	}
```

However, a proper fix would be required since I am not a Go developer and do not know why this logger statement would hang or if this repo is using the Go SDK properly.

**Comment by ffahri** [comment]
Thank you @xenalite, I will look into this

**Comment by ffahri** [comment]
@xenalite @WhiskyTest still try to reproduce this issue
i created 150+ resources and still could not reproduced yet

https://gist.github.com/ffahri/197df8fe4a981c138c041d643b8f4c9f

v:0.3.4


edit: added 30+ schedules still no luck about reproducing 

teams 
```
  resource "opsgenie_team" "test**" {
    name        = "sorry**"
    description = "This team deals with all the things"
  }
```
schedules
```
  

  resource "opsgenie_schedule" "test**" {
    name        = "geniescheduletest-**" 
    description = "schedule test"
    timezone    = "Europe/Rome"
    enabled     = false
}
 
```

**Comment by ffahri** [comment]
@xenalite if the creating the logger on the our side does this fix your case? if so we can implement, could you confirm 
``` 
func (c *Config) Client() (*OpsgenieClient, error) {
	config := &client.Config{
		ApiKey:         c.ApiKey,
		RetryCount:     10,
		OpsGenieAPIURL: client.ApiUrl(c.ApiUrl),
		Backoff:        retryablehttp.DefaultBackoff,
		Logger: logrus.New(),
	}
```

**Comment by ffahri** [comment]
Closing this issue, if this still persist please re-open

**Comment by wwilson** [comment]
This issue seems to be affecting us. (Opsgenie provider version 0.6.5). Oddly, it reproduces perfectly deterministically on one computer, but never affects others. We've deleted all local terraform state and confirmed that the machines are running byte-for-byte identical versions of terraform _and_ the provider, but the issue persists. 

We've tried enabling TRACE-level debug logging from terraform, but haven't gotten anything out of that besides more evidence that it's a deadlock in the opsgenie provider.

Did you all figure out anything further about this bug?
